website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,678 @@
+
+![Form field example](/images/form-field.png 'Form field example')
+
+A field component consists of an input element wrapped with other field elements such as label, description, error message, hint, badge, etc.
+
+Depending on the use case, you might want to use a date field, a text field, or a select field. In the UIKit library, we provide a variety of components to cover different use cases. For more information about field and input components check their related packages in the [commercetools/ui-kit](https://github.com/commercetools/ui-kit) repository.
+
+<Info>
+
+All UIKit field components have a related input component, for example `<TextField>` -> `<TextInput>`.
+
+</Info>
+
+## Accessibility support
+
+All field and input components have built-in support for accessibility features such as aria labels, keyboard navigation and focus management, error messaging, and more.
+
+<Info>
+
+Testing these field components with [React Testing Library](https://github.com/testing-library/react-testing-library) can be easily done using selectors such as `*ByLabelText`, `*ByRole`, etc.
+
+</Info>
+
+# Form validation
+
+One of the important parts of forms is validation. It is important to validate constraints of forms such as required fields but also to check additional semantic requirements, for example checking that the value is a valid URL.
+
+Asides from client-side validation, forms can also perform asynchronous validation against an API to ensure data correctness before submitting the form.
+
+For that, Formik allows you to implement a [`validate` function](https://formik.org/docs/api/formik#validate-values-values--formikerrorsvalues--promiseany) that should return an object of errors.
+
+```ts
+type TFieldErrors = Record<string, boolean>;
+// Similar shape of `FormikErrors` but values are `TFieldErrors` objects.
+type TCustomFormErrors<Values> = {
+  [K in keyof Values]?: TFieldErrors;
+};
+
+declare const validate = (values: FormValues) => TCustomFormErrors<FormValues>;
+```
+
+Specifically, the object returned from the `validate` function should contain properties correlating to field names with their values being objects with reasons for the given error. Let's take a look at an example:
+
+```js title="validate.js"
+import LocalizedTextInput from '@commercetools-uikit/localized-text-input';
+
+const validate = (values) => {
+  const errors = {};
+
+  if (LocalizedTextInput.isEmpty(values.name)) {
+    errors.name = { missing: true };
+  }
+
+  return errors;
+}
+```
+
+In this example, we are validating that the `name` field has a required value. If the value is empty (no localized values have been provided), we assign to the `errors.name` property an error object with the error key `missing` set to `true`.
+
+In the field component, you need to make sure to assign the `errors` and `touched` props which will make the field component render an error message, in case one was returned from the `validate` function.
```

Comment:
```suggestion
In the field component, you must assign the `errors` and `touched` props which make the field component render an error message, in case one was returned from the `validate` function.
```